sources/packages/backend/apps/api/src/services/application-restriction-bypass/application-restriction-bypass.service.ts

```diff
@@ -49,4 +82,380 @@ export class ApplicationRestrictionBypassService extends RecordDataModelService<
+            "applicationRestrictionBypass.studentRestriction = studentRestriction.id",
+          )
+          .andWhere("applicationRestrictionBypass.isActive = true");
+        return `NOT EXISTS (${subQuery.getQuery()})`;
+      })
+      // Restriction action type condition.
+      .andWhere((qb) => {
+        const actionTypeSubQuery = qb
+          .subQuery()
+          .select("1")
+          .from(Application, "application")
+          .innerJoin("application.currentAssessment", "currentAssessment")
+          .innerJoin("currentAssessment.offering", "offering")
+          .where("application.id = :applicationId", { applicationId })
+          .andWhere(
+            new Brackets((qb) => {
+              qb.where(
+                new Brackets((fullTimeQb) => {
+                  fullTimeQb
+                    .where("offering.offeringIntensity = :fullTimeIntensity", {
+                      fullTimeIntensity: OfferingIntensity.fullTime,
+                    })
+                    .andWhere(
+                      "restriction.action_type::text[] && ARRAY[:...fullTimeActionTypes]::text[]",
+                      {
+                        fullTimeActionTypes: [
+                          RestrictionActionType.StopFullTimeBCFunding,
+                          RestrictionActionType.StopFullTimeDisbursement,
+                        ],
+                      },
+                    );
+                }),
+              ).orWhere(
+                new Brackets((partTimeQb) => {
+                  partTimeQb
+                    .where("offering.offeringIntensity = :partTimeIntensity", {
+                      partTimeIntensity: OfferingIntensity.partTime,
+                    })
+                    .andWhere(
+                      "restriction.action_type::text[] && ARRAY[:...partTimeActionTypes]::text[]",
+                      {
+                        partTimeActionTypes: [
+                          RestrictionActionType.StopPartTimeDisbursement,
+                        ],
+                      },
+                    );
+                }),
+              );
+            }),
+          );
+        return `EXISTS (${actionTypeSubQuery.getQuery()})`;
+      })
+      .andWhere("studentRestriction.isActive = true")
+      .orderBy("restriction.restrictionCode", "ASC")
+      .getMany();
+  }
+
+  /**
+   * Creates a new application restriction bypass.
+   * @param payload application restriction bypass data.
+   * @param auditUserId id of the user who is creating the bypass.
+   * @returns created application restriction bypass.
+   */
+  async bypassRestriction(
+    payload: BypassRestrictionAPIInDTO,
+    auditUserId: number,
+  ): Promise<ApplicationRestrictionBypass> {
+    const auditUser = { id: auditUserId } as User;
+    const now = new Date();
+    return await this.dataSource.transaction(
+      async (transactionalEntityManager) => {
+        await this.checkForActiveApplicationStudentRestrictionByPasses(
```

Comment:
Is there a benefit to executing these validations inside the transition?
If one of them fails a transaction does not even need to be started.